### PR TITLE
website/docs: flow context docs

### PR DESCRIPTION
**Pull request by BeryJu**
document flow context variables, when they are set and what they do

### CODE REVIEW

File: website/docs/flow/context/index.md [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+Stores the final redirect that the user's browser will be sent to after the flow is finished executing successfully. This is set when an un-authenticated user attempts to access a secured application, and when a user authenticates/enrolls with an external source.
+
+### `application` (Application object)
+
+When an unauthenticated user attempts to access a secured resource, they are redirected to an authentication flow. The application they attempted to access will be stored in this key.
+
+### `source` (Source object)
+
+When a user authenticates/enrolls via an external source, this will be set to the source they are using.
```
Comment (tanberry) [inline website/docs/flow/context/index.md:41]: Will there be a key attached to the source object? Like `source.SAML`

What do we mean by "source" here, and in general?

File: website/docs/flow/context/index.md [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+
+## Scenario specific
```
Comment (tanberry) [inline website/docs/flow/context/index.md:43]:
```suggestion
## Scenario-specific objects
```

Comment (tanberry) [inline website/docs/flow/context/index.md:43]:
```suggestion
## Scenario-specific objects
```

File: website/docs/flow/context/index.md [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+
+### `pending_user` (User object)
+
+`pending_user` is used by multiple stages. In the context of most flow executions, it holds the user that is executing the flow. This value is not set automatically, it is set via the [Identification stage](../stages/identification/).
+
+Stages that require a user, such as the [Password stage](../stages/password/), the [Authenticator validation stage](../stages/authenticator_validate/) and others will use this value if it is set, and fallback to the request's users when possible.
+
+### `prompt_data` (Dictionary)
+
+`prompt_data` is primarily used by the [Prompt stage](../stages/prompt/). The value of any field within a prompt stage is written to the `prompt_data` dictionary. For example, given a field with the _Field key_ `email` that was submitted with the value `foo@bar.baz` will result in the following context:
```
Comment (tanberry) [inline website/docs/flow/context/index.md:19]:
```suggestion
`prompt_data` is primarily used by the [Prompt stage](../stages/prompt/). The value of any field within a prompt stage is written to the `prompt_data` context (the dictionary of data used in that flow). For example, given a field with the _Field key_ `email` that was submitted with the value `foo@bar.baz` will result in the following context:
```

File: website/docs/flow/context/index.md [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+
+An optional list of all permissions that will be given to the application by granting consent. Not supported with SAML. When used with an OAuth2 or Proxy provider, this will be set based on the configured scopes.
+
+### Autosubmit stage
+
+The autosubmit stage is an internal stage type that is not configurable via the API/Web interface. It is used in certain situations, where a POST request has to be sent from the browser, such as with SAML POST bindings. This works by using an HTML form that is submitted automatically.
+
+#### `title` (string)
+
+Optional title of the form shown to the user. Automatically set when this stage is used by the backend.
```
Comment (tanberry) [inline website/docs/flow/context/index.md:67]: Automatically set to what value? Does user ever see it?

File: website/docs/flow/context/index.md [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+
+URL that the form will be submitted to.
+
+#### `attrs` (dict)
+
+Key-value pairs of the data that is included in the form and will be submitted to `url`.
+
+### User write stage
+
+#### `groups` (List of Group objects)
```
Comment (tanberry) [inline website/docs/flow/context/index.md:79]:
```suggestion
#### `groups` (list of Group objects)
```

File: website/docs/flow/context/index.md [hunk 6]
```diff
@@ -0,0 +1,129 @@
+---
+title: Flow Context
+---
+
+Each flow execution has an independent context. This context can hold arbitrary data, which can be used and transformed by stages and policies.
+
+Any data can be stored in the flow context, however there are some reserved keys in the context dictionary that are used by authentik stages.
```
[the remark this answers is not in the document]
Comment (tanberry) [reply website/docs/flow/context/index.md:7]: Oh. I learned after writing this note that this page basically is the dictionary.

**Review by tanberry — COMMENTED** [review]
Awesome new content, need to add a bit more conceptual info about what contexts are and how they work.

File: website/docs/flow/context/index.md [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+
+## Common objects
+
```
[the remark this answers is not in the document]
Comment (BeryJu) [reply website/docs/flow/context/index.md:19]: Dictionary is also a data-type (https://www.programiz.com/python-programming/dictionary) 

Good point, especially the `prompt_data` dictionary has some specific keys that translate to user properties

File: website/docs/flow/context/index.md [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+```json
+{
+    "prompt_data": {
+        "email": "foo@bar.baz"
+    }
+}
+```
+
+This data can be modified with policies. The data is also used by stages like [User write](../stages/user_write.md), which takes data in `prompt_data` and writes it to `pending_user`.
```
[the remark this answers is not in the document]
Comment (BeryJu) [reply website/docs/flow/context/index.md:38]: No, all of these can be modified with policies

File: website/docs/flow/context/index.md [hunk 9]
```diff
@@ -0,0 +1,129 @@
...
+
+### `redirect` (string)
+
+Stores the final redirect that the user's browser will be sent to after the flow is finished executing successfully. This is set when an un-authenticated user attempts to access a secured application, and when a user authenticates/enrolls with an external source.
```
[the remark this answers is not in the document]
Comment (BeryJu) [reply website/docs/flow/context/index.md:42]: the final redirect URL that the user is redirected to after the flow is done

File: website/docs/flow/context/index.md [hunk 10]
```diff
@@ -0,0 +1,129 @@
...
+
+### `application` (Application object)
+
+When an unauthenticated user attempts to access a secured resource, they are redirected to an authentication flow. The application they attempted to access will be stored in this key.
```
[the remark this answers is not in the document]
Comment (BeryJu) [reply website/docs/flow/context/index.md:46]: The key is always just `application` with the value being the application object

File: website/docs/flow/context/index.md (hunk 1 above)
Comment (BeryJu) [reply website/docs/flow/context/index.md:50]: same as above

Sources are https://goauthentik.io/docs/terminology#source, i.e. federation sources, like log into authentik with github

File: website/docs/flow/context/index.md (hunk 4 above)
Comment (BeryJu) [reply website/docs/flow/context/index.md:76]: Usually in a format of `Redirecting to $app...`. The user sees it very briefly as the redirect happens

This stage is a bit different as an admin can't configure it via the API/web interface, it's an "in-memory" stage that gets added by authentik itself when it's required

File: website/docs/flow/context/index.md [hunk 11]
```diff
@@ -0,0 +1,172 @@
...
+
+Stores the final redirect URL that the user's browser will be sent to after the flow is finished executing successfully. This is set when an un-authenticated user attempts to access a secured application, and when a user authenticates/enrolls with an external source.
+
+#### Identifications tage
+
+##### `pending_user_identifier` (string)
+
+If _Show matched user_ is disabled, this key will hold the user identifier entered by the user in the identification stage.
+
+Stores the final redirect that the user's browser will be sent to after the flow is finished executing successfully. This is set when an un-authenticated user attempts to access a secured application, and when a user authenticates/enrolls with an external source.
```
Comment (tanberry) [inline website/docs/flow/context/index.md:56]:
```suggestion
Stores the final redirect URL that the user's browser will be sent to after the flow is finished executing successfully. This is set when an un-authenticated user attempts to access a secured application, and when a user authenticates/enrolls with an external source.
```

**Review by tanberry — APPROVED** [review]
LGTM!

**Pull request merged**